packages/firebase_ml/lib/src/firebase_model_download_conditions.dart

```diff
@@ -0,0 +1,19 @@
+/// Conditions to download remote models.
+class FirebaseModelDownloadConditions {
+  /// Boolean value that indicates if wifi is required.
+  final bool requireWifi;
+
+  /// Boolean value that indicates if device idle is required.
+  final bool requireDeviceIdle;
+
+  /// Boolean value that indicates if charging is required.
+  final bool requireCharging;
+
+  /// Constructor for the download conditions that takes optional parameters
```

Comment:
nit: The first sentence should be a summary on one line. [style guide](https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph)